### ISSUE TITLE
Matches does not work with uppercase fields

### ISSUE THREAD

**Issue by TetianaPozniakova** [bbyars/mountebank#228]
#### Expected behaviour

When a request matches regex , mb should return corresponding response.

#### Actual behaviour

Mb does not return body when field starts with uppercase
![image](https://cloud.githubusercontent.com/assets/3495360/25043745/d6e9ab46-212a-11e7-8815-de111954851d.png)

but does return when all letters are lowercase:
![image](https://cloud.githubusercontent.com/assets/3495360/25043820/7787c114-212b-11e7-9e7e-8998531c9163.png)

#### Steps to reproduce

1.  Add Imposter:
`{
  "port": 5000,
  "protocol": "http",
  "name": "DefaultMock",
  "stubs": [
    {
      "predicates": [
        {
          "matches": {
            "body": {
              "Data": "^\\\"UserId\\\":\\\"14352c82-d189-45cd-bb13-051a86b80764\\\"$",
              "TTL": "^(\\d{4})-(\\d{2})-(\\d{2})T(\\d{2})(:\\d{2}){2}.(\\d+)[+-](\\d{2}):(\\d{2})$"
            }
          }
        }
      ],
      "responses": [
        {
          "is": {
            "statusCode": 200,
            "body": "UserProfileNotFound"
          }
        }
      ]
    }
  ]
}`

2. Post request:
`{
"Data":"\"UserId\":\"14352c82-d189-45cd-bb13-051a86b80764\"",
"TTL":"2017-04-13T13:00:33.7696005+00:00"
}`
3. Mb does not return expected body.
P.S. Running mb via docker

#### Software versions used

```
OS         : alpine:3.5
mountebank : 1.9 and 1.10

Dockerfile:
```
FROM alpine:3.5

ENTRYPOINT ["mb"]
CMD ["start"]
ENV MOUNTEBANK_VERSION=1.10.0

RUN apk add --update nodejs
RUN npm install -g mountebank@${MOUNTEBANK_VERSION} --production
```
```

#### Log contents in mb.log when running mb --loglevel debug

```
keyvaluestore_1          | info: [mb:2525] POST /imposters/
keyvaluestore_1          | debug: [mb:2525] ::ffff:172.27.0.1:44202 => {"port":5000,"protocol":"http","name":"DefaultMock","stubs":[{"predicates":[{"matches":{"body":{"Data":"^\\\"UserId\\\":\\\"14352c82-d189-45cd-bb13-051a86b80764\\\"$","TTL":"^(\\d{4})-(\\d{2})-(\\d{2})T(\\d{2})(:\\d{2}){2}.(\\d+)[+-](\\d{2}):(\\d{2})$"}}}],"responses":[{"is":{"statusCode":200,"body":"UserProfileNotFound"}}]},{"predicates":[{"matches":{"body":{"Data":"14352c82-d189-45cd-bb13-051a86b80764"}}}],"responses":[{"is":{"statusCode":200,"body":"TTL"}}]}]}
keyvaluestore_1          | info: [http:5000 DefaultMock] Open for business...
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 ESTABLISHED
keyvaluestore_1          | info: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 => POST /
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 => {"requestFrom":"::ffff:172.27.0.1:34516","method":"POST","path":"/","query":{},"headers":{"Host":"keyvaluestore:5300","Connection":"keep-alive","Content-Length":"46","Postman-Token":"92ad42d2-00ad-f2a6-dc6e-ef5d01350900","Cache-Control":"no-cache","Origin":"chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop","User-Agent":"Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36","Content-Type":"text/plain;charset=UTF-8","Accept":"*/*","Accept-Encoding":"gzip, deflate","Accept-Language":"en-US,en;q=0.8,ru;q=0.6"},"body":"{\n\"data\":\"2017-04-13T13:00:33.7696005+00:00\"\n}"}
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 no predicate match
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 generating response from {"is":{}}
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 <= {"statusCode":200,"headers":{"Connection":"close"},"body":"","_mode":"text"}
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34516 CLOSED
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34518 ESTABLISHED
keyvaluestore_1          | info: [http:5000 DefaultMock] ::ffff:172.27.0.1:34518 => POST /
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34518 => {"requestFrom":"::ffff:172.27.0.1:34518","method":"POST","path":"/","query":{},"headers":{"Host":"keyvaluestore:5300","Connection":"keep-alive","Content-Length":"49","Postman-Token":"029b181d-ca5b-b4c4-48a0-6aa321b5356a","Cache-Control":"no-cache","Origin":"chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop","User-Agent":"Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36","Content-Type":"text/plain;charset=UTF-8","Accept":"*/*","Accept-Encoding":"gzip, deflate","Accept-Language":"en-US,en;q=0.8,ru;q=0.6"},"body":"{\n\"Data\":\"14352c82-d189-45cd-bb13-051a86b80764\"\n}"}
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34518 no predicate match
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34518 generating response from {"is":{}}
keyvaluestore_1          | debug: [http:5000 DefaultMock] ::ffff:172.27.0.1:34518 <= {"statusCode":200,"headers":{"Connection":"close"},"body":"","_mode":"text"}
```

**Comment by TetianaPozniakova** [comment]
It worked in mb 1.7 but not after that.

**Comment by bbyars** [comment]
Thanks for all the detail. Just fixed; will release with v1.11 in the next couple days.
